pkgs/applications/editors/emacs/elisp-packages/update

```diff
@@ -1,8 +1,9 @@
 #!/usr/bin/env bash
 set -euo pipefail
 
-SCRIPT_DIR="$( cd "$(dirname "$0")" ; pwd -P )"
-cd "$SCRIPT_DIR"
+SCRIPT_DIRECTORY=$(cd "$(dirname "${BASH_SOURCE[@]}")"; cd -P "$(dirname "$(readlink "${BASH_SOURCE[@]}" || echo .)")"; pwd)
```

Comment:
Is all these necessary?  For example, `readlink` probably is not needed here.

Reply:
> Is all these necessary? For example, readlink probably is not needed here.

Check the [post](https://www.binaryphile.com/bash/2020/01/12/determining-the-location-of-your-script-in-bash.html) that @AndersonTorres linked to in #402168, the rationale for the whole thing is explained there in detail.

Reply:
I did not read that post, but I guess the gist is that `SCRIPT_DIRECTORY=$(cd "$(dirname "${BASH_SOURCE[@]}")"; cd -P "$(dirname "$(readlink "${BASH_SOURCE[@]}" || echo .)")"; pwd)` handles **every** case.  My concern is there are some unnecessary complexity for **this case**.

Reply:
> handles **every** case.

To be precise, it handles *almost* every case (excluding chained symlinks).